server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java

```diff
@@ -2047,7 +2047,15 @@ public void testElectionSchedulingAfterDiscoveryOutage() {
                     + 4 * DEFAULT_DELAY_VARIABILITY
                     // Then a commit of the new leader's first cluster state
                     + DEFAULT_CLUSTER_STATE_UPDATE_DELAY
-                    // Then the remaining node may join
+                    // Then the remaining node may experience a disconnect (see comment in PeerFinder#closePeers) for which it is
+                    // removed from the cluster, and its fault detection must also detect its removal
+                    + Math.max(
+                        DEFAULT_CLUSTER_STATE_UPDATE_DELAY,
```

Comment:
Not sure if I understood/remember it correctly from when I took a look, but I remember there was one task still pending when the check here would time out. Was the failing case due to the disconnected node causing the new cluster state update to not succeed and step down again, adding another round?

Reply:
No all the cluster state updates succeed, it's just that one of them removes a node, then that node does 3 leader checks before trying to rejoin. The total timeout was less than 4s before this change, and the 3 leader checks alone could take up to 3.9s.